### PR TITLE
Enable parallel computing for find_center_vo, allow 2D sinogram input…

### PR DESCRIPTION
**Pull request by nghia-vo**
…, fix bug caused by downsampling

### CODE REVIEW

File: source/tomopy/recon/rotation.py [hunk 1]
```diff
@@ -203,14 +204,14 @@ def _find_center_cost(
 
 
 def find_center_vo(tomo, ind=None, smin=-50, smax=50, srad=6, step=0.25,
-                   ratio=0.5, drop=20):
+                   ratio=0.5, drop=20, parallel=True):
```
Comment (carterbox) [inline source/tomopy/recon/rotation.py:207]: Please match the rest of the library by using `ncore, nchunk` parameters with default value of `None`. The `distribute_jobs` function automatically chooses values for these parameters.

https://github.com/tomopy/tomopy/blob/1047c0b59de3bf5026209b670ba4b1ebd03ba9f9/source/tomopy/prep/stripe.py#L88-L90

Comment (carterbox) [inline source/tomopy/recon/rotation.py:207]: Please match the rest of the library by using `ncore, nchunk` parameters with default value of `None`. The `distribute_jobs` function automatically chooses values for these parameters.

https://github.com/tomopy/tomopy/blob/1047c0b59de3bf5026209b670ba4b1ebd03ba9f9/source/tomopy/prep/stripe.py#L88-L90

File: source/tomopy/recon/rotation.py [hunk 2]
```diff
@@ -250,49 +256,111 @@ def find_center_vo(tomo, ind=None, smin=-50, smax=50, srad=6, step=0.25,
...
-        fft2sino = np.fft.fftshift(fft2(sino_sino))
-        np.abs(fft2sino, out=abs_fft2_sino)
-        abs_fft2_sino *= mask
-        list_metric[i - smin] = abs_fft2_sino.mean()
+    if parallel is True:
+        list_metric = distribute_jobs(np.float32(list_shift),
+                                      _calculate_metric, axis=0,
+                                      args=(sino, flip_sino, comp_sino, mask),
+                                      ncore=mp.cpu_count() - 1, nchunk=1)
+    else:
```
Comment (carterbox) [inline source/tomopy/recon/rotation.py:349]: For example, instead of setting `parallel=False`, users can set `ncore=1` and `nchunk=1`.

**Review by carterbox — CHANGES_REQUESTED** [review]
Hi @nghia-vo. thanks for the PR. In addition to the comments below, could you please separate the changes for the bug fix and the "allowing of 2D inputs" into their own commits? It will help me understand which changes go with which objective.

**Comment by nghia-vo** [comment]
Update and separate commits as requested. However nchunk must be set to 1 for the function to work.

**Comment by carterbox** [comment]
I think it's fine if `nchunk=1` is hardcoded in this case. I also removed the `if ncore !=1 else` statement. The number of cores should not affect the result of `distribute_jobs()`.

**Comment by nghia-vo** [comment]
I've checked the update. Good to go.

**Pull request merged**